File: src/algo/sort/01.bubbleSort.js

function bubbleSort(arr) {
    var len = arr.length;
    for( var i = 0; i < len -1; i++) {
        var flag = true;
        for(var j = 0; j < len - 1 - i; j++) {
            if(arr[j] > arr[j+1]) {
                var temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;

                flag = false;
            }
        }
        
        if(flag) {
            break;
        }
    }

}

var arr = [1,10, 100,20,39,70, 69];
bubbleSort(arr);
console.log(arr);

File: src/algo/sort/01.bubbleSort.py

def bubbleSort(arr):
    length = len(arr)
    for i in range(1, length):
        flag = True
        for j in range(0, length - i):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
                flag = False
        if(flag):
            break
    return arr

arr = [1,3,20,9,60,49,78,200]
bubbleSort(arr)
print(arr)
